libraries/chain/apply_context.cpp

```diff
@@ -95,27 +121,12 @@ void apply_context::exec_one()
                } catch( const wasm_exit& ) {}
             }
 
-            if( !privileged && control.is_builtin_activated( builtin_protocol_feature_t::ram_restrictions ) ) {
-               const size_t checktime_interval = 10;
-               size_t counter = 0;
-               bool not_in_notify_context = (receiver == act->account);
-               const auto end = _account_ram_deltas.end();
-               for( auto itr = _account_ram_deltas.begin(); itr != end; ++itr, ++counter ) {
-                  if( counter == checktime_interval ) {
-                     trx_context.checktime();
-                     counter = 0;
-                  }
-                  if( itr->delta > 0 && itr->account != receiver ) {
-                     EOS_ASSERT( not_in_notify_context, unauthorized_ram_usage_increase,
-                                 "unprivileged contract cannot increase RAM usage of another account within a notify context: ${account}",
-                                 ("account", itr->account)
-                     );
-                     EOS_ASSERT( has_authorization( itr->account ), unauthorized_ram_usage_increase,
-                                 "unprivileged contract cannot increase RAM usage of another account that has not authorized the action: ${account}",
-                                 ("account", itr->account)
-                     );
-                  }
+            if (!privileged) {
+               if (control.is_builtin_activated(builtin_protocol_feature_t::ram_restrictions)) {
+                  check_unprivileged_resource_usage<unauthorized_ram_usage_increase>("RAM", _account_ram_deltas);
                }
+
+               check_unprivileged_resource_usage<unauthorized_disk_usage_increase>("DISK", _account_disk_deltas);
```

Comment:
I believe everything "DISK" related is not supposed to be a part of this release, so I think this param can be removed.

Reply:
> I believe everything "DISK" related is not supposed to be a part of this release, so I think this param can be removed.

The disk charging part and its corresponding unit tests existed before this PR.  This PR only refactor the code so that the permission checking mechanism for DISK is the same with that for RAM. Unless we don't want to charge for DISK at all in the future. I don't see why we need to remove the logic.